MeetU/MeetU/Scripts/Directives/meetup-entry/meetup-entry.js

```diff
@@ -0,0 +1,118 @@
+﻿(function () {
+    "use strict";
+    angular
+        .module("meetupModule")
+        .directive('meetupEntry', function () {
+            var controller = function() {
+                var vm = this;
+
+                init();
+
+                function init() {
+                    var data = angular.copy(vm.datasource),
+                        meetup = data.meetup,
+                        joins = data.joinViews,
+                        i, len, currentPerson;
+
+                    vm.meetup = {
+                        title: meetup.title,
+                        description: meetup.description,
+                        when: convertTime(meetup.when),
+                        createdAt: convertTime(meetup.createdAt),
+                        id: meetup.id,
+                        sponsorId: meetup.sponsor,
+                        sponsorUserName: data.sponsorUserName,
+                        sponsorNickName: data.sponsorNickName,
+                        updatedAt: convertTime(meetup.updatedAt),
+                        where: meetup.where,
+                    };
+
+                    vm.meetup.status = vm.meetup.when.expired ? "Expired" : "Join";
+                    vm.meetup.colorBackClass = vm.meetup.when.expired ? "expired-back" : "";
+                    vm.meetup.colorFrontClass = vm.meetup.when.expired ? "expired-front" : "";
+                    vm.meetup.colorTxtClass = vm.meetup.when.expired ? "expired-txt" : "";
+                    vm.meetup.colorMeridiem = vm.meetup.when.meridiem === "AM" ? "when-am" : "when-pm";
+
+                    vm.joins = {
+                        joinedNumber: joins.length,
+                        joinedPeople: []
+                    };
+
+                    for (i = 0, len = joins.length; i < len; ++i) {
+                        currentPerson = {};
+                        currentPerson.name = joins[i].theJoinedUserNickName || joins[i].join.userName;
+                        currentPerson.id = joins[i].join.userId;
+                        currentPerson.joinedAt = convertTime(joins[i].join.at).completeTimeString;
+                        vm.joins.joinedPeople.push(currentPerson);
+                    }
+
+
+
+                }
+
+
+
+                function convertTime(timeString) {
```

Comment:
this is quite complicated, and it will generate more bugs in the future. I am just wondering if there is a mature lib can do this for us.